app/src/main/java/com/sunasterisk/music_72/screen/adapter/GenreAdapter.kt

```diff
@@ -0,0 +1,75 @@
+package com.sunasterisk.music_72.screen.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.view.animation.AnimationUtils
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.recyclerview.widget.AsyncDifferConfig
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.sunasterisk.music_72.R
+import com.sunasterisk.music_72.data.model.Genre
+import com.sunasterisk.music_72.databinding.LayoutItemHomeBinding
+import com.sunasterisk.music_72.screen.viewmodel.ItemGenreViewModel
+import com.sunasterisk.music_72.utils.OnRecyclerViewItemListener
+import kotlinx.android.synthetic.main.layout_item_home.view.*
+import java.util.concurrent.Executors
+
+class GenreAdapter() : ListAdapter<Genre, GenreAdapter.ViewHolder>(
+    AsyncDifferConfig.Builder<Genre>(GenreDiffCallback())
+        .setBackgroundThreadExecutor(Executors.newSingleThreadExecutor())
+        .build()
+) {
+    private var listener: OnRecyclerViewItemListener<Genre>? = null
+
+    fun setOnItemClickListener(listener: OnRecyclerViewItemListener<Genre>?) {
+        this.listener = listener
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        return ViewHolder(DataBindingUtil.inflate(
+            LayoutInflater.from(parent.context),
+            R.layout.layout_item_home,
+            parent,
+            false
+            ), listener
+        )
```

Comment:
viết cẩn thận loại đoạn này nhé 
Đừng cố gắng viết tắt 
Nếu nó cồng kềnh, sử dụng các biến phụ để tách dòng ra cho clear 
```suggestion
        val inflater = LayoutInflater.from(parent.context)
        val binding = DataBindingUtil.inflate(inflater, R.layout.layout_item_home, parent, false)
        return ViewHolder(binding, listener)
```